addons/portal_rating/static/src/js/portal_rating_composer.js

```diff
@@ -61,12 +62,39 @@ publicWidget.registry.RatingPopupComposer = publicWidget.Widget.extend({
      * @override
      */
     start: function () {
-        var ratingPopupData = this.$el.data();
-        var ratingPopup = new RatingPopupComposer(this, ratingPopupData);
+        this.ratingPopupData = this.$el.data();
+        this.ratingPopupData.display_composer = this.ratingPopupData.disable_composer ? false : !session.is_website_user;
+        this.ratingPopup = new RatingPopupComposer(this, this.ratingPopupData);
+        this.ratingPopup.on('reload_composer_widget', null, this._reloadRatingPopupComposerWidget.bind(this));
         return Promise.all([
             this._super.apply(this, arguments),
-            ratingPopup.appendTo(this.$el)
+            this.ratingPopup.appendTo(this.$el)
         ]);
     },
+    _reloadRatingPopupComposerWidget: function (newMessage) {
+        // destroy existing ratingPopup
+        let data = newMessage.data;
+        var oldComposer = this.ratingPopup;
+        if (oldComposer) {
+            oldComposer.destroy();
+            // load last message value for portal chatter review
+            core.bus.trigger('post_message', newMessage);
+        }
+        // instanciate and insert ratingPopup widget
+        if (this.ratingPopupData['display_composer']) {
+            if (newMessage !== undefined) {
```

Comment:
This condition is useless. You wrote

```javascript
let data = newMessage.data
```

on the first line. If newMessage is undefined, the function has already crashed.